src/script/descriptor.cpp

```diff
@@ -524,6 +540,21 @@ class RawDescriptor final : public DescriptorImpl
 public:
     RawDescriptor(CScript script) : DescriptorImpl({}, {}, "raw"), m_script(std::move(script)) {}
     bool IsSolvable() const final { return false; }
+
+    Optional<OutputType> GetOutputType() const override
+    {
+        CTxDestination dest;
+        assert(ExtractDestination(m_script, dest));
```

Comment:
This seems a bit aggressive, maybe return `nullopt` is `ExtractDestination` fails?

Reply:
Reverted the change. There's no need to check the result of `ExtractDestination` because it is directly implied by the resulting `CTxDestination`. If it fails, that will be `CNoDestination`.